### DISCUSSION TITLE
type hint for dataframe

### DISCUSSION THREAD

**Discussion by stevenlis** [pandas-dev/pandas-stubs#426] [Q&A] [answered]
I wonder if this is a "standard" way or if the community has a consensus for the type hint of dataframe.
There seems to be a couple of ways mentioned in the following link
https://stackoverflow.com/questions/43890844/pythonic-type-hints-with-pandas

Including the following two:
```python
import pandas as pd

def do_something(df: pd.DataFrame) -> pd.DataFrame:
    # do something
    return df
```
or 
```python
from pandas import DataFrame

def do_something(df: DataFrame) -> DataFrame:
    # do something
    return df
```

I realized that there is no doc for pandas-stubs yet, but how should I write the type hint for DataFrame now?

**Comment by gandhis1** [comment, marked as answer]
I think this is as subjective as allowing `from` imports or not. Most people I have seen tend to do the `pd` import alias, so the former is probably most common. But it's hardly a matter of right or wrong, it's completely subjective.

**Reply by stevenlis** [reply]
Sorry for the confusion.🤪 I was not concerned about relative vs absolute import. I just wonder if using `pd.DataFrame` is commonly accepted. If you click the SO link, it seems like there are many ways to annotate a DataFrame type.
